packages/cozy-codemods/src/transforms/replace-icons-svgr.js

```diff
@@ -11,64 +11,58 @@ export default function replaceSvgrIcons(file, api) {
   const j = api.jscodeshift
   const root = j(file.source)
 
+  const transform = path => {
+    const iconAttr = path.node.attributes.find(
+      attr => attr.name && attr.name.name === 'icon'
+    )
+
+    if (iconAttr && iconAttr.value.type === 'Literal') {
+      const componentName = iconNameToComponentName(iconAttr.value.value)
+
+      imports.ensure(
+        root,
+        {
+          default: `${componentName}Icon`
+        },
+        `cozy-ui/transpiled/react/Icons/${componentName}`
+      )
+      iconAttr.value = j.jsxExpressionContainer(
+        j.jsxIdentifier(`${componentName}Icon`)
+      )
+    }
+  }
+
   root
     .find(j.JSXOpeningElement, {
       name: {
-        name: 'Icon'
+        name: 'Avatar'
       }
     })
-    .forEach(path => {
-      const iconAttr = path.node.attributes.find(
-        attr => attr.name.name === 'icon'
-      )
+    .forEach(transform)
 
-      if (iconAttr.value.type === 'Literal') {
-        const componentName = iconNameToComponentName(iconAttr.value.value)
-
-        imports.ensure(
-          root,
-          {
-            default: `${componentName}Icon`
-          },
-          `cozy-ui/transpiled/react/Icons/${componentName}`
-        )
-        iconAttr.value = j.jsxExpressionContainer(
-          j.jsxIdentifier(`${componentName}Icon`)
-        )
+  root
+    .find(j.JSXOpeningElement, {
+      name: {
+        name: 'Icon'
       }
     })
+    .forEach(transform)
 
   root
     .find(j.JSXOpeningElement, {
       name: {
         name: 'Button'
       }
     })
-    .forEach(path => {
-      const iconAttr = path.node.attributes.find(
-        attr => attr.name && attr.name.name === 'icon'
-      )
+    .forEach(transform)
 
-      if (iconAttr && iconAttr.value.type === 'Literal') {
-        const componentName = iconNameToComponentName(iconAttr.value.value)
-
-        imports.ensure(
-          root,
-          {
-            default: `${componentName}Icon`
-          },
-          `cozy-ui/transpiled/react/Icons/${componentName}`
-        )
-        imports.ensure(
-          root,
-          {
-            default: `Icon`
-          },
-          `cozy-ui/transpiled/react/Icon`
-        )
-        iconAttr.value = `{<Icon icon={${componentName}Icon} />}`
+  root
+    .find(j.JSXOpeningElement, {
+      name: {
+        name: 'ButtonLink'
```

Comment:
On peut encore simplifier en : 

const supportedComponent = ['Avatar', 'Button', 'ButtonLink', 'Icon']
for (let component of supportedComponent) {
root.find(j.JSXOpeningElement, { name: { name: component }}). .forEach(transform)
}

comme ça c'est clair et précis les cas qu'on supporte et il est facile d'en rajouter si besoin ?

Reply:
C'est fait merci, au début je pensais que les cas allaient différer un peu, mais au final tout est pareil maintenant.